src/reducers/ast.js

```diff
@@ -207,6 +207,32 @@ export function getPausePoints(
   return state.ast.pausePoints.get(sourceId);
 }
 
+export function getFirstPausePointLocation(
+  state: OuterState,
+  location: ?Location
+): ?Location {
+  if (!location) {
+    return;
+  }
+
+  const pausePoints = getPausePoints(state, location.sourceId);
+  if (!pausePoints) {
+    return location;
+  }
+
+  const pausesAtLine = pausePoints[String(location.line)];
+  if (pausesAtLine) {
+    const column = Object.keys(pausesAtLine).find(
```

Comment:
would Object.values be simpler?

Reply:
No, because I wouldn't have access to the key (line number), which is what I need.